cpp/src/copying/slice.cu

```diff
@@ -215,9 +215,7 @@ public:
       }
 
       if (output_column->dtype == GDF_STRING_CATEGORY){
-        NVCategory* new_category = static_cast<NVCategory*>(input_column_.dtype_info.category)->gather_and_remap(
-                      static_cast<int*>(output_column->data), (unsigned int)output_column->size);
-        output_column->dtype_info.category = new_category;
+        CUDF_TRY(nvcategory_gather(output_column, static_cast<NVCategory*>(input_column_.dtype_info.category)));
```

Comment:
Why is `slice` being changed in this PR?

Reply:
Because I updated nvcategory_utils.cu 
```diff
-    NVStrings* tptr = static_cast<NVCategory*>(column->dtype_info.category)->to_strings();
+    NVStrings* tptr = static_cast<NVCategory*>(column->dtype_info.category)->gather_strings(static_cast<nv_category_index_type*>(column->data),
+                                                                                            column->size,
+                                                                                            DEVICE_ALLOCATED);
```
with `to_strings` I was getting the wrong order, but slice unit test started crashing, changing to `nvcategory_gather` fixed that which also, I think, is the correct function instead of `gather_and_remap`